components/ws-manager-api/go/config/config.go

```diff
@@ -453,10 +494,62 @@ func RenderWorkspacePortURL(urltpl string, ctx PortURLContext) (string, error) {
 	return b.String(), nil
 }
 
-// ResourceConfiguration configures resources of a pod/container
-type ResourceConfiguration struct {
+// ResourceRequestConfiguration configures resources of a pod/container
+type ResourceRequestConfiguration struct {
 	CPU              string `json:"cpu"`
 	Memory           string `json:"memory"`
 	EphemeralStorage string `json:"ephemeral-storage"`
 	Storage          string `json:"storage,omitempty"`
 }
+
+type ResourceLimitConfiguration struct {
+	CPU              *CpuResourceLimit `json:"cpu"`
```

Comment:
Any reason why we don't set omitempty to all fields?

Reply:
I just kept it consistent with the previous behavior. It is also nice to not have type out the fields when editing the configmap in e.g. preview environments.